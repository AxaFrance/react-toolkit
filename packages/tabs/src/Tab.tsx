import React from 'react';
import {
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  compose,
} from '@axa-fr/react-toolkit-core';

interface TabComponentProps {
  className?: string;
  title: string;
  children?: React.ReactNode;
  classModifier?: string;
}

const Tab: React.SFC<TabComponentProps> = () => <span />;

const DEFAULT_CLASSNAME = 'af-tabs__pane';

export type TabProps = TabComponentProps & WithClassModifierOptions;

const enhance = compose<TabComponentProps, TabProps>(
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
);

const Enhanced = enhance(Tab);
Enhanced.displayName = 'TabCore';
export default Enhanced;
