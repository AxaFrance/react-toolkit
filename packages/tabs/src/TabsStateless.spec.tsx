import React from 'react';
import { create } from 'react-test-renderer';
import TabStateless from './TabsStateless';
import Tab from './Tab';

const noop = () => {};
describe('<TabStateless>', () => {
  it('should render component', () => {
    const component = create(
      <TabStateless
        onChange={noop}
        className="className"
        activeIndex="1"
        classModifier="withIcon">
        <Tab title="title tab1" classModifier="modifier 1">
          <span>Content</span>
        </Tab>
        <Tab title="title tab2" classModifier="modifier">
          <span>Content</span>
        </Tab>
        <Tab title="title tab3" classModifier="modifier 1">
          <span>Content</span>
        </Tab>
      </TabStateless>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
