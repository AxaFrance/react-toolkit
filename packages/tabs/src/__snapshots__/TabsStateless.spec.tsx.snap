// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabStateless> should render component 1`] = `
<div
  className="className className--withIcon"
>
  <ul
    className="af-tabs__control"
  >
    <li
      className="af-tabs__item af-tabs__item--modifier af-tabs__item--1"
    >
      <button
        className="af-tabs__link"
        id="0"
        onClick={[Function]}
      >
        title tab1
      </button>
    </li>
    <li
      className="af-tabs__item af-tabs__item--modifier af-tabs__item--active"
    >
      <button
        className="af-tabs__link"
        id="1"
        onClick={[Function]}
      >
        title tab2
      </button>
    </li>
    <li
      className="af-tabs__item af-tabs__item--modifier af-tabs__item--1"
    >
      <button
        className="af-tabs__link"
        id="2"
        onClick={[Function]}
      >
        title tab3
      </button>
    </li>
  </ul>
  <div
    className="af-tabs__content"
  >
    <div
      className="af-tabs__pane"
    >
      <span>
        Content
      </span>
    </div>
  </div>
</div>
`;
